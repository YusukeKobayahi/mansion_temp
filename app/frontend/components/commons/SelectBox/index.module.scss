@import "~/styles/variables";
@import "~/styles/utils";

.selectBox {
  width: 100%;
  height: 50px;

  select {
    width: 100%;
    height: 100%;
    appearance: none;
    color: #746744;
    font-size: 14px;
    font-weight: bold;
    background-color: #ffffff;
    padding: 0 8px;
    cursor: pointer;
    background-image: url("/ms-library/images/select.png");
    background-size: 12px 10px;
    background-position: right 5px center;
    background-repeat: no-repeat;
    border: 1px solid #767676;
  }
}
