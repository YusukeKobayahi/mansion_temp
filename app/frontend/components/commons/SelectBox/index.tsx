import styles from "~/components/commons/SelectBox/index.module.scss";
import classnames from "classnames";

type SelectBoxProps = {
  defaultValue: string;
  options: {
    value: string;
    text: string;
  }[];
  onBlur?: React.FocusEventHandler<HTMLSelectElement>;
  onChange?: React.FocusEventHandler<HTMLSelectElement>;
  className?: string;
};

const SelectBox: React.FC<SelectBoxProps> = ({
  options,
  defaultValue,
  onBlur,
  onChange,
  className = "",
}: SelectBoxProps) => {
  const optionsNode = options.map((option, i) => {
    const { value, text } = option;
    return (
      <option key={i} value={value}>
        {text}
      </option>
    );
  });
  return (
    <div className={classnames(styles.selectBox, className)}>
      <select onBlur={onBlur} onChange={onChange} defaultValue={defaultValue}>
        {optionsNode}
      </select>
    </div>
  );
};

export default SelectBox;
