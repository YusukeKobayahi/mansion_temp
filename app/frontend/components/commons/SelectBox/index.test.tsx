import { render } from "@testing-library/react";
import SelectBox from "~/components/commons/SelectBox";

describe("components/commons/SelectBox", () => {
  const options = [10, 20, 30].map((num) => {
    return {
      value: String(num),
      text: String(num),
    };
  });
  it("正常に表示される", () => {
    const { asFragment } = render(
      <SelectBox options={options} defaultValue={""} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
