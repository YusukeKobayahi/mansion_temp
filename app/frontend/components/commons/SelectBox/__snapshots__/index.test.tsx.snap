// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/commons/SelectBox 正常に表示される 1`] = `
<DocumentFragment>
  <div
    class="selectBox"
  >
    <select>
      <option
        value="10"
      >
        10
      </option>
      <option
        value="20"
      >
        20
      </option>
      <option
        value="30"
      >
        30
      </option>
    </select>
  </div>
</DocumentFragment>
`;
